relay-server/src/utils/metrics_rate_limits.rs

```diff
@@ -38,7 +38,12 @@ impl<Q: AsRef<Vec<Quota>>> MetricsLimiter<Q> {
     /// Create a new limiter instance.
     ///
     /// Returns Ok if `metrics` contain transaction metrics, `metrics` otherwise.
-    pub fn create(buckets: Vec<Bucket>, quotas: Q, scoping: Scoping) -> Result<Self, Vec<Bucket>> {
+    pub fn create(
+        buckets: Vec<Bucket>,
+        quotas: Q,
+        scoping: Scoping,
+        usage: bool,
```

Comment:
`usage` may be hard to understand outside this scope. What do you think about making this an enum of two variants (usage and transaction duration)?

Reply:
Agreed. The metrics rate limiter overall has a convoluted signature, so I chose the simplest change to follow up with a proper refactoring. I'm not yet sure about whether this should be an enum, the config, or something else, so I'd prefer to stick with the simplest choice. Would that work?